app/middlewares/checkAuthentication.js

```diff
@@ -24,3 +29,15 @@ module.exports.validate = (req, res, next) => {
 
   return next();
 };
+
+module.exports.validateAdmin = (req, res, next) =>
+  decode(req.headers.token)
+    .then(user => {
+      if (user.type !== 'admin') {
```

Comment:
'admin' should not be harcoded here. Add a constants file and add 'ADMIN' and 'USER' there, and make it clear both are user types.
Also, we prefer to write constants like this ALL_CAPS to better diferentiate them.